### PR TITLE
Enhance test_backend_test.py and simplify backend test CI

### PR DESCRIPTION
**Pull request by jcwchen**
**Description**
- Use `onnx/test/test_backend_test.py` to fully test all backend data (use strict shape_inference)
- Update invalid models: Constant-1 does not allow int64 and it will fail shape_inference; bump the opset_version into Constant-9
  - test_PixelShuffle
  - test_operator_repeat
  - test_operator_repeat_dim_overflow
- Reuse test_backend_test.py to do the backend node test data check; Remove test_generated_backend.py

**Motivation**
test_backend_test.py has the same ability as `workflow_scripts/test_generated_backend.py `. ONNX should simplify backend test CI by enhancing existing test_backend_test.py.

### CODE REVIEW

File: .github/workflows/manylinux/test_package_i686.sh
Comment (askhade) [inline .github/workflows/manylinux/test_package_i686.sh:30]: what is the motivation behind generating the data in the CI pipeline? why not simply test the data which is already checked in?

Comment (jcwchen) [reply .github/workflows/manylinux/test_package_i686.sh:30]: For now, it can only make sure models from test generation on various platforms (Windows, Linux, Mac) pass strict onnx.checker. It does not require the models generated from different platforms to be the same as the checked-in ones. We can probably achieve that after https://github.com/onnx/onnx/pull/3855 is merged.

File: .github/workflows/release_win.yml
Comment (askhade) [inline .github/workflows/release_win.yml:115]: what is "-k OnnxBackendNodeModelTest" for?

Comment (askhade) [reply .github/workflows/release_win.yml:115]: Also earlier test script was running both checker and shape_inference checks on models and now we are only doing the shape_inference check?

File: .github/workflows/manylinux/test_package_i686.sh
Comment (askhade) [reply .github/workflows/manylinux/test_package_i686.sh:30]: I think we should wait for #3855 and then refine this PR. 
Once #3855 is merged we should use since_version to create all the test data and run a git diff check to verify there are no diffs. This will help us validate the PR is including all the test data that it needs to (based on the unit test cases added) and that the test data is refreshed (in the past we have run into cases where author's made changes to the test case and forgot to regenerate the test data. This resulted in bad test data being checked in)

File: .github/workflows/release_win.yml
Comment (jcwchen) [reply .github/workflows/release_win.yml:115]: Specify OnnxBackendNodeModelTest here will make onnx\test\test_backend_test only test those models under `onnx/backend/test/case/node/`.

> Also earlier test script was running both checker and shape_inference checks on models and now we are only doing the shape_inference check?

Yes. To keep the same behavior as previous verification, I just added onnx.checker in test_backend_test.py. It can also enhance the test for backend data.

File: onnx/test/test_backend_test.py
Comment (askhade) [inline onnx/test/test_backend_test.py:46]: I suggest moving check_model above infer_shapes. No point in inferring shapes for an invalid model.

File: .github/workflows/release_win.yml
Comment (askhade) [reply .github/workflows/release_win.yml:115]: Where is "OnnxBackendNodeModelTest " defined? The name suggests it will run tests for both Node and Model

Comment (jcwchen) [reply .github/workflows/release_win.yml:115]: I cannot find the source code about it shortly, but from my understanding the name here is collected as "OnnxBackend-[directory name under [here](https://github.com/onnx/onnx/tree/master/onnx/backend/test/data)]-ModelTest".
For instance, OnnxBackend"Node"ModelTest or OnnxBackend"PyTorchConverted"ModelTest.

**Review by liqunfu — APPROVED** [review]

**Comment by jcwchen** [comment]
@onnx/sig-operators-approvers Could you please review/approve this PR? Thank you for your time!

**Review by gramalingam — APPROVED** [review]

**Pull request merged**
